test/index.spec.js

```diff
@@ -125,16 +125,90 @@ describe('Testing index', () => {
     }).timeout(10000);
   });
 
+  it("Testing Multi Version Query", async () => {
+    const indexName = "version-index-test";
+    const meta = { mappings: { idx: { properties: { uuid: { type: "keyword" } } } } };
+
+    // initialization
+    await index.rest.call("DELETE", `${indexName}@*`);
+    const create1 = await index.rest.call('PUT', `${indexName}@1`, { body: meta });
+    expect(create1.statusCode).to.equal(200);
+    const create2 = await index.rest.call('PUT', `${indexName}@2`, { body: meta });
+    expect(create2.statusCode).to.equal(200);
+
+    // add data
+    const id1 = uuid4();
+    await index.rest.call('PUT', `${indexName}@2/idx/${id1}`, { body: { uuid: id1 } });
+    await index.rest.call('PUT', `${indexName}@1/idx/${id1}`, { body: { uuid: id1 } });
+    await index.rest.call("POST", `${indexName}@*`, { endpoint: "_refresh" });
+
+    // run query
+    const result = await index.rest.call('GET', `${indexName}@*`, { endpoint: "_search" });
+    expect(result.body.hits.total).to.equal(2);
+
+    // cleanup
+    const delResult = await index.rest.call("DELETE", `${indexName}@*`);
+    expect(delResult.statusCode).to.equal(200);
+  });
+
   describe('Testing REST interaction', () => {
+    const validate = async (count, historic) => {
+      expect(await index.rest.data.refresh("offer")).to.equal(true);
+      expect(await index.rest.data.count("offer")).to.equal(count);
+      expect(await index.rest.mapping.historic("offer")).to.deep.equal(historic);
+    };
+    const checkDocs = async (uuids) => {
+      expect(await index.rest.data.query("offer", index.query.build("offer", {
+        toReturn: ["id"],
+        filterBy: { and: [["id", "in", uuids]] },
+        limit: 1,
+        offset: 1
+      }))).to.deep.equal({
+        payload: [{ id: uuids[1] }],
+        page: {
+          next: { limit: 1, offset: 2 },
+          prev: { limit: 1, offset: 0 },
+          max: 3,
+          cur: 2,
+          size: 1
+        }
+      });
+    };
+
+    it("Testing versioning", async () => {
+      const uuids = [uuid4(), uuid4(), uuid4()].sort();
+      await index.rest.mapping.delete("offer");
+      // create new index
+      expect(await index.rest.mapping.create("offer")).to.equal(true);
+      await validate(0, {});
+      // insert data
+      expect(await index.rest.data.update("offer", { upsert: uuids.map(id => ({ id })) })).to.equal(true);
+      await validate(3, {});
+      // create new version of index
+      index.index.register("offer", Object.assign({}, indices.offer, { fields: ["id"] }));
+      expect(await index.rest.mapping.create("offer")).to.equal(true);
+      await validate(3, { "offer@409492a32436d2cec4a3d01046308ae7e53893f1": 3 });
```

Comment:
This hash is going to change whenever we update the offer index, right? Would it make sense to calculate this in the setup? `currentOfferHash`

Reply:
It will. I'm too lazy to do that now. Let's see how often this breaks :)

Reply:
I lied. Fixing now